proposals/body-tracking.html

```diff
@@ -0,0 +1,182 @@
+          onEndSession: onEndSession,
+          textEnterXRTitle: "START AR",
+          textXRNotFoundTitle: "AR NOT FOUND",
+          textExitXRTitle: "EXIT AR",
+        });
+
+        document.querySelector('header').appendChild(xrButton.domElement);
+
+        if (navigator.xr) {
+          navigator.xr.isSessionSupported('immersive-ar')
+                      .then((supported) => {
+            xrButton.enabled = supported;
+          });
+        }
+
+        window.addEventListener( 'resize', onWindowResize );
+
+      }
+
+      function onRequestSession() {
+        let sessionInit = {
+          requiredFeatures: ['body-tracking'],
+          optionalFeatures: ['local-floor', 'bounded-floor'],
+        };
+        navigator.xr.requestSession('immersive-ar', sessionInit).then((session) => {
+          session.mode = 'immersive-ar';
+          xrButton.setSession(session);
+          onSessionStarted(session);
+        });
+      }
+
+      function onSessionStarted(session) {
+        session.addEventListener('end', onSessionEnded);
+
+        renderer.xr.setSession(session);
+
+        renderer.setAnimationLoop(render);
+      }
+
+      function onEndSession(session) {
+        session.end();
+      }
+
+      function onSessionEnded(event) {
+        xrButton.setSession(null);
+
+        renderer.setAnimationLoop(null);
+        renderer.xr.setSession(null)
+      }
+
+      function onWindowResize() {
+        camera.aspect = window.innerWidth / window.innerHeight;
+        camera.updateProjectionMatrix();
+
+        renderer.setSize(window.innerWidth, window.innerHeight);
+      }
+
+
+      function render(timestamp, frame) {
+        if (frame) {
+
+          if ( frame.body ) {
+
+            let body = frame.body;
+
+            if (spheres == undefined) {
+
+              const geometry = new THREE.IcosahedronGeometry( 0.01, 3 );
+              const material = new THREE.MeshLambertMaterial();
+
+              spheres = new THREE.InstancedMesh( geometry, material, body.size );
+              spheres.translateZ( -1 ).setRotationFromMatrix (new THREE.Matrix4().makeRotationY(Math.PI));
+              spheres.instanceMatrix.setUsage( THREE.DynamicDrawUsage ); // will be updated every frame
+              scene.add( spheres );
+
+              const color = new THREE.Color();
+
+              for ( let i = 0; i < spheres.count; i ++ ) {
+
+                spheres.setColorAt( i, color.setHex( 0xffffff * Math.random() ) );
+
+              }
+
+              spheres.instanceMatrix.needsUpdate = true;
+
+            }
+
+            let i = 0;
+            const matrix = new THREE.Matrix4();
+            let space = renderer.xr.getReferenceSpace();
+            body.forEach(part => {
+
+              const pose = frame.getPose(part, space);
+              const position = pose.transform.position;
+
+              if (!part.jointName.includes("hand")) {
+                matrix.copy(scalehand);
```

Comment:
Why do hands need to be scaled up 3x? In your video it looks like the opposite, and the hands are scaled down?

Reply:
Oh, wait, I see. This is checking to ensure that "hand" *isn't* part of the name. That's pretty confusing. Can you invert the logic here?